opengever/setup/sections/fileloader.py

```
@@ -0,0 +1,109 @@
+from opengever.document.document import IDocumentSchema
+from opengever.document.subscribers import set_digitally_available
+from opengever.document.subscribers import sync_title_and_filename_handler
+from opengever.setup.sections.jsonsource import BUNDLE_PATH_KEY
+from opengever.setup.sections.jsonsource import JSON_STATS_KEY
+from zope.annotation.interfaces import IAnnotations
+from zope.interface import classProvides
+from zope.interface import implements
+import logging
+import os.path
+
+
+logger = logging.getLogger('opengever.bsmigration.fileloader')
+
+
+class FileLoaderSection(object):
+    """Stores file data directly in already constructed content items."""
+
+    classProvides(ISectionBlueprint)
+    implements(ISection)
+
+    def __init__(self, transmogrifier, name, options, previous):
+        self.previous = previous
+        self.context = transmogrifier.context
+
+        # TODO: Might want to also use defaultMatcher for this key to make it
+        # configurable instead of hard coding it here.
+        self.key = 'filepath'
+        self.pathkey = defaultMatcher(options, 'path-key', name, 'path')
+
+        self.bundle_path = IAnnotations(transmogrifier)[BUNDLE_PATH_KEY]
+
+        # TODO: Handle .eml / ftw.mail.mail
+        self.file_field_class = IDocumentSchema['file']._type
+
+        self.stats = IAnnotations(transmogrifier)[JSON_STATS_KEY]
+        self.stats['errors']['files_not_found'] = {}
+        self.stats['errors']['files_io_errors'] = {}
+        self.stats['msgs'] = {}
+
+    def __iter__(self):
+        for item in self.previous:
+            keys = item.keys()
+            pathkey = self.pathkey(*keys)[0]
+
+            if self.key in item:
+                filepath = item[self.key]
+                if filepath is None:
+                    yield item
+                    continue
+
+                if pathkey not in item:
+                    logger.warning("Missing path key for file %s" % filepath)
```

we might want to let a validator section check if the file exists and fail hard here instead